src/Http/Response/Factory.php

```diff
@@ -7,8 +7,8 @@
 use Illuminate\Support\Str;
 use Dingo\Api\Http\Response;
 use Illuminate\Support\Collection;
-use Illuminate\Pagination\Paginator;
 use Dingo\Api\Transformer\Factory as TransformerFactory;
+use Illuminate\Contracts\Pagination\LengthAwarePaginator;
```

Comment:
No... It shouldn't as the `LengthAwarePaginator` contract extends the `Paginator` contract. So even if you pass in an instance of `Illuminate\Pagination\LengthAwarePaginator` it should be fine as it is, through inheritance, an instance of `Illuminate\Contracts\Pagination\Paginator`.